src/fields/commonFieldTypes.ts

```diff
@@ -11,8 +11,14 @@ export interface CommonFieldProps extends MarginProps {
   required?: boolean
   error?: boolean
   errorMsg?: string
+  status?: {
+    type: Status
+    showStatus: boolean
+  }
```

Comment:
I think this is a bit overly complicated?
the `showStatus` is not needed cause it can be derived by the optional `status` property and at that point there's no real need of having a `type` you can just have `status?: 'complete' | ...`

Reply:
So it's made that way to avoid layout shift and allow for animation. 
If `status` is truthy everything renders so that there is additional space for the status
And the showStatus actually triggers the animation. 

We could have it as simple as `status?: 'complete' | ...` but then is there a way to animate it / make sure there is no layout shift each time the status comes in?

Reply:
we have layout shift with errors as well right now and I don't think it's ever been an issue. It's impossible to pre-render something that doesn't exist cause the other issue is that if it's alway there you have an extra gap of white space that's a problem as much as the layout shift

Reply:
you can potentially try to always render it and play around with opacity and visibility in order to make it animate avoiding the white space gap